Mathlib/CategoryTheory/Limits/Preserves/Presheaf.lean

```diff
@@ -0,0 +1,190 @@
+not end up in situations where we have morphisms like `colimit.ι F i ≫ f`, where the domain of `f`
+is `colimit G` where `F` and `G` are definitionally equal but not syntactically equal. In these
+situations, lemmas about `colimit.ι G i ≫ f` cannot be applied using `rw` and `simp`, forcing the
+use of `erw`. In particular, for us this means that `HasColimit.isoOfNatIso (Iso.refl _)` is better
+than `Iso.refl _` and that we should be very particular about functor associativity.
+
+The theorem is also true for large categories and the proof given here generalizes to this case on
+paper. However, our infrastructure for commuting finite limits of shape `J` and filtered colimits
+of shape `K` is fundamentally not equipped to deal with the case where not all limits of shape `K`
+exist. In fact, not even the definition `colimitLimitToLimitColimit` can be written down if not
+all limits of shape `K` exist. Refactoring this to the level of generality we need would be a major
+undertaking. Since the application to the category of Ind-objects only require the case where `C`
+is small, we leave this as a TODO.
+
+## References
+* [M. Kashiwara, P. Schapira, *Categories and Sheaves*][Kashiwara2006], Proposition 3.3.13
+* [F. Borceux, *Handbook of Categorical Algebra 1*][borceux-vol1], Proposition 6.1.2
+-/
+
+open CategoryTheory Limits
+
+universe v u
+
+namespace CategoryTheory.Limits
+
+section LargeCategory
+
+variable {C : Type u} [Category.{v} C] [HasFiniteColimits C] (A : Cᵒᵖ ⥤ Type v)
+
+/-- If `C` is a finitely cocomplete category and `A : Cᵒᵖ ⥤ Type u` is a presheaf that preserves
+finite limites, then `CostructuredArrow yoneda A` is filtered.
+
+One direction of Proposition 3.3.13 of [Kashiwara2006].
+-/
+theorem isFiltered_costructuredArrow_yoneda_of_preservesFiniteLimits
+    [PreservesFiniteLimits A] : IsFiltered (CostructuredArrow yoneda A) := by
+  suffices IsFiltered A.Elementsᵒᵖ from
+    IsFiltered.of_equivalence (CategoryOfElements.costructuredArrowYonedaEquivalence _)
+  suffices IsCofiltered A.Elements from isFiltered_op_of_isCofiltered A.Elements
+  suffices HasFiniteLimits A.Elements from IsCofiltered.of_hasFiniteLimits A.Elements
+  exact ⟨fun J _ _ => inferInstance⟩
+
+end LargeCategory
+
+variable {C : Type u} [SmallCategory C] [HasFiniteColimits C]
+
+variable (A : Cᵒᵖ ⥤ Type u)
+
+namespace PreservesFiniteLimitsOfIsFilteredCostructuredArrowYonedaAux
+
+variable {J : Type} [SmallCategory J] [FinCategory J] (K : J ⥤ Cᵒᵖ)
+
+/-- (Implementation) This is the bifunctor we will apply "filtered colimits commute with finite
+limits" to. -/
+def functorToInterchange : J ⥤ CostructuredArrow yoneda A ⥤ Type u :=
+  K ⋙ coyoneda ⋙ (whiskeringLeft _ _ _).obj (CostructuredArrow.proj _ _)
+
+/-- (Implementation) The definition of `functorToInterchange`. -/
+@[simps!]
+def functorToInterchangeIso : functorToInterchange A K ≅
+    K ⋙ coyoneda ⋙ (whiskeringLeft _ _ _).obj (CostructuredArrow.proj _ _) :=
+  NatIso.ofComponents (fun _ => Iso.refl _)
```

Comment:
```suggestion
  Iso.refl _
```

Reply:
Similarly for the next two isos. Is there a reason to use `NatIso.ofComponents` here?

Reply:
There are places where you don't want a natural isomorphism to hold by refl, but I agree that this is not one of them.